File: python/job.py
class Job:
    def __init__(self, name, after, before):
        self.name = name
        self.after = after
        self.before = before

    def __init__(self, key, value):
        self.name = key
        self.after = value['after'] if 'after' in value else []
        self.before = value['before'] if 'before' in value else []

    def __str__(self):
        result_template =  'Job: name=\'{name}\',after=[{after}],before=[{before}]'
        result = result_template.format(name=self.name, after=self.after, before=self.before)
        return result

File: cpp/src/job.cpp
#include "job.h"

namespace algorithm
{

job::job(const std::set<std::string>& after, const std::set<std::string>& before) :
        m_after(after),
        m_before(before)
{
}

bool job::operator==(const job& other) const
{
    return (m_after == other.m_after) &&
            (m_before == other.m_before);
}

void job::add_after(const std::string& job_after)
{
    m_after.insert(job_after);
}

void job::add_before(const std::string& job_before)
{
    m_before.insert(job_before);
}

const std::set<std::string>& job::get_after() const
{
    return m_after;
}

const std::set<std::string>& job::get_before() const
{
    return m_before;
}

} // namespace algorithm

